internal/models/accounts_test.go

```diff
@@ -1,13 +1,285 @@
 package models_test
 
 import (
+	"encoding/json"
 	"testing"
+	"time"
 
+	"github.com/formancehq/go-libs/v3/pointer"
 	"github.com/formancehq/payments/internal/models"
+	"github.com/google/uuid"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestToPSPAccount(t *testing.T) {
 	assert.Nil(t, models.ToPSPAccount(nil))
 	assert.NotNil(t, models.ToPSPAccount(&models.Account{}))
 }
+
+func TestFromPSPAccount(t *testing.T) {
+	t.Parallel()
+
+	now := time.Now().UTC()
+	connectorID := models.ConnectorID{
+		Reference: uuid.New(),
+		Provider:  "test",
+	}
+
+	t.Run("valid account", func(t *testing.T) {
+		t.Parallel()
+		// Given
+		pspAccount := models.PSPAccount{
+			Reference: "acc123",
+			CreatedAt: now,
+			Name:      pointer.For("Test Account"),
+			DefaultAsset: pointer.For("USD/2"),
+			Metadata: map[string]string{
+				"key": "value",
+			},
+			Raw: json.RawMessage(`{"test": "data"}`),
+		}
+
+		account, err := models.FromPSPAccount(pspAccount, models.ACCOUNT_TYPE_INTERNAL, connectorID)
+		
+		// Then
+			require.NoError(t, err)
```

Comment:
Thoughout the PR, there are these kind of indentation issue. Please check indentation, specially after a comment.